### DISCUSSION TITLE
How to update an existing model and extend it with new photos

### DISCUSSION THREAD

**Discussion by Trebol-a** [ultralytics/ultralytics#16483] [General]
I have a trained yolov8 (called bigmodel by example) model with quite good success. There are 70 classes trained with approximately 80k photos. This training took 15 days to reach 190 epochs.

In some cases however the classification of some classes fails, so I prepare a new dataset with 400-500 photos of the worst classifying images from 8-9 classes only. 

So far what I have tried is to do:

`yolo detect train model=‘bigmodel.pt’ data=‘bigmodel.yaml’ epochs=50 imgsz=512 device=0 workers=8 batch=16 project=‘trainings’ name=‘bigmodel-upgrade-01’ pretrained=True`

I keep the original YAML file which still defines 70 classes, although the dataset only provisions 500 photos of 8-9 classes. Then the problem is that the new ‘upgraded’ model (called bigmodel-upgrade-01 by example) tends to confuse classes by seeing many ‘upgraded’ classes where there are none.

What would be the right way to ‘update/expand’ my previously bigmodel trained model without having to retrain everything?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Trebol-a, thank you for reaching out and sharing your use case with us 🌟! We're thrilled to hear about your progress with YOLOv8. To help you further, please note:

If this is a 🐛 Bug Report, ensure that you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better understand and replicate the issue.

Your question about updating a model is great for our community! If you haven’t already, check our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal results when training on your new dataset.

An Ultralytics engineer will also assist you soon with specific guidance. In the meantime, feel free to explore our discussion platforms:

- Real-time chat on [Discord](https://ultralytics.com/discord) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com)
- Join our [Subreddit](https://reddit.com/r/ultralytics) for community knowledge exchange

## Upgrade

Ensure you’re using the latest `ultralytics` package, which might have improvements or fixes:

```bash
pip install -U ultralytics
```

## Environments

Run YOLOv8 in one of our supported environments with all dependencies pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it means all our [CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) are passing, verifying the correct operation across multiple platforms.

Stay tuned, and we appreciate your patience while we assist you 🚀.

**Comment by glenn-jocher** [comment]
@Trebol-a to update your model without retraining everything, consider using transfer learning. Fine-tune your existing model with the new dataset by setting `pretrained=True` and adjusting the learning rate to be lower. Ensure your dataset and YAML file are consistent with the classes you're focusing on.
